File: .config/config/ags/widgets/powerbutton/powerbutton.js
import { powerMenuOptions } from "./powermenu.js";
const hyprland = await Service.import("hyprland");


export const PowerButton = () =>
  Widget.Button({
    class_name: "power-button",
    on_clicked: () =>{
        if(powerMenuOptions.reveal_child){
            powerMenuOptions.reveal_child = false
            Utils.timeout(400, () => {
                App.toggleWindow("powermenu")
            })
        } else{
            App.toggleWindow("powermenu")
            powerMenuOptions.reveal_child = true
        }
            
    },
    child: Widget.Icon({
      class_name: "power-button-icon",
      icon: "system-shutdown-symbolic",
    }),
    tooltip_text: "Power Menu",
  });

File: Everblush/.config/fabric/widgets/powerbutton.py
from fabric.widgets.button import Button
from utils import add_cursor_hover
from window_layers.powermenu.powermenu import PowerMenu


def PowerButton():
    powermenu = PowerMenu()
    button = Button(
        name="power-button",
        label="",
        on_clicked=lambda *_: powermenu.toggle_powermenu(),
    )
    add_cursor_hover(button)
    return button
